src/Serializer/JsonApiSerializer.php

```diff
@@ -572,4 +553,54 @@ private function addIncludedDataToRelationship($includeObject, $relationship)
 
         return $relationship;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function injectAvailableIncludeData($data, $availableIncludes)
+    {
+        if (!$this->shouldIncludeLinks()) {
+            return $data;
+        }
+
+        if ($this->isCollection($data)) {
+            $data['data'] = array_map(function ($resource) use ($availableIncludes) {
+                foreach ($availableIncludes as $relationshipKey) {
+                    $resource = $this->addRelationshipLinks($resource, $relationshipKey);
+                }
+                return $resource;
+            }, $data['data']);
+        } else {
+            foreach ($availableIncludes as $relationshipKey) {
+                $data['data'] = $this->addRelationshipLinks($data['data'], $relationshipKey);
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Adds links for all available includes to a single resource.
+     *
+     * @param array $resource         The resource to add relationship links to
+     * @param string $relationshipKey The resource key of the relationship
+     */
+    private function addRelationshipLinks($resource, $relationshipKey)
```

Comment:
Would it make sense to make this method protected? I can see cases where it would be nice to have a custom link structure here.